test/test_ipca.py

```diff
@@ -0,0 +1,74 @@
+# Copyright Contributors to the Cellarium project.
+# SPDX-License-Identifier: BSD-3-Clause
+
+import lightning.pytorch as pl
+import numpy as np
+import pytest
+import torch
+
+from scvid.module import IncrementalPCA
+from scvid.train import TrainingPlan
+
+from .common import TestDataset
+
+n, g = 10000, 100
+
+
+@pytest.fixture
+def x_ng():
+    rng = torch.Generator()
+    rng.manual_seed(1465)
+    mean_g = torch.randn((g,), generator=rng)
+    x_ng = torch.randn((n, g), generator=rng) + mean_g
+    return x_ng
+
+
+@pytest.mark.parametrize("mean_correct", [False, True])
+@pytest.mark.parametrize("batch_size", [10_000, 5000, 1000, 500, 250])
+@pytest.mark.parametrize("k", [30, 50, 80])
+def test_incremental_pca(x_ng: np.ndarray, mean_correct: bool, batch_size: int, k: int):
```

Comment:
Don't we want to check `p_oversamples`, i.e. that the error decreases with increased `p_oversamples`?  Also, how about checking the results against exact SVD?  Neither is necessary but they'd make good tests.